R/cv_varsel.R

```diff
@@ -68,22 +68,45 @@
 #' @inherit varsel details return
 #'
 #' @note If `validate_search` is `FALSE`, the search is not included in the CV
-#'   so that only a single full-data search is run.
+#'   so that only a single full-data search is run. If the number of observations
+#'   is big, the fast PSIS-LOO-CV along the full-data search path is likely
+#'   to be accurate. If the number of observations is small or moderate,
+#'   the fast PSIS-LOO-CV along the full-data search path is likely to have
+#'   optimistic bias in the middle of the search path. This result can be
+#'   used to guide further actions and the optimistic bias can be greatly
+#'   reduced by using `validate_search=TRUE`.
 #'
-#'   For PSIS-LOO CV, \pkg{projpred} calls [loo::psis()] (or, exceptionally,
+#'   PSIS uses Pareto-\eqn{\hat{k}} diagnostic to assess the reliability of PSIS-LOO-CV.
+#'   Whether the Pareto-\eqn{\hat{k}} diagnostics are shown as warnings, is controlled
+#'   with a global option `projpred.warn_psis` (default is FALSE).
+#'   See [loo::loo-glossary] for how to interpret the Pareto-\eqn{\hat{k}} values and
+#'   the warning thresholds. \pkg{projpred} does not support the usually recommended
+#'   moment-matching (see [loo::loo_moment_match()] and [brms::loo_moment_match()]),
+#'   mixture importance sampling (`vignette("loo2-mixis", package="loo")`),
+#'   or `reloo`-ing ([brms::reloo()]). If the reference model PSIS-LOO-CV
+#'   Pareto-\eqn{\hat{k}} values are good, but there are high Pareto-\eqn{\hat{k}}
+#'   values for the projected models, you can try increasing the number of draws used
+#'   for the PSIS-LOO-CV (`ndraws_pred` with  `refit_prj=TRUE`). If increasing the
+#'   number of draws does not help and if the reference model PSIS-LOO-CV
+#'   Pareto-\eqn{\hat{k}} values are high, and the PSIS-LOO-CV results change substantially
+#'   when using moment-matching, mixture importance sampling, or `reloo`-ing, we
+#'   recommend to use $K$-fold-CV within `projpred`.
+#' 
+#'   For PSIS-LOO-CV, \pkg{projpred} calls [loo::psis()] (or, exceptionally,
 #'   [loo::sis()], see below) with `r_eff = NA`. This is only a problem if there
 #'   was extreme autocorrelation between the MCMC iterations when the reference
 #'   model was built. In those cases however, the reference model should not
 #'   have been used anyway, so we don't expect \pkg{projpred}'s `r_eff = NA` to
 #'   be a problem.
 #'
-#'   PSIS cannot be used if the draws have different (i.e., nonconstant) weights
```

Comment:
This part about nonconstant weights does not appear in the new text anymore. Was this on purpose? As far as I know, `loo::psis()` is not able to deal with nonconstant weights out-of-the-box.

Reply:
If the draws have weights, we can just multiply them with importance weights and use Pareto diagnostics. This approach was used, e.g., in http://proceedings.mlr.press/v97/magnusson19a.html. I made the change in the code in that other PR, but forgot that the doc changes are also there.

Reply:
I see. So with PR #496, PSIS for draws with nonconstant weights will be supported. Then I'll resolve this conversation here.